src/main/scala/org/broadinstitute/clio/service/AuthorizationService.scala

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.clio.service
+
+import akka.http.scaladsl.server.RequestContext
+import org.broadinstitute.clio.model._
+
+
+case class AuthorizationInfo(token: String, expiration: String, email: String, id: String)
+
+class AuthorizationService() {
+
+  def authorize(context: RequestContext): Boolean = {
+    val pairs = for (header <- context.request.headers) yield
+      header match {
+        case OidcAccessToken(token)      => 0 -> token
+        case OidcClaimExpiresIn(expires) => 1 -> expires
+        case OidcClaimEmail(email)       => 2 -> email
+        case OidcClaimUserId(id)         => 3 -> id
+        case OidcClaimSub(id)            => 3 -> id
+      }
+    (pairs.size == 4) && (Vector.tabulate(4) { pairs.toMap } match {
+      case Vector(token, expires, email, id) =>
+        AuthorizationService.mock == AuthorizationInfo(token, expires, email, id)
```

Comment:
I'm unclear what `.mock` is used for here. I would expect it to be a mocked service API for testing but it looks like it contains the actual values from the HTTP response header.

It looks like this is changing the value in `AuthorizationService.mock` but I don't see how that could work since it's declared as a `val`.